### DISCUSSION TITLE
Date validation not working

### DISCUSSION THREAD

**Discussion by HepplerDotNet** [livewire/livewire#3036] [7. Bugs (archived)]
In one of my components I validate a date to be sure it's not in the past.

```php
 protected $rules = [
        'account.name' => 'required',
        'account.email' => 'required|email',
        'account.valid_until' => 'date|after_or_equal:today',
    ];
    protected $messages = [
        'account.name.required' => 'Name should not be empty',
        'account.email.required' => 'E-Mail should not be empty',
        'account.email.email' => ':input is no valid E-Mail',
        'account.valid_until.date' => ':input is not a valid date',
        'account.valid_until.after_or_equal' => ':input is in the past',
    ];
```
On the Account model i cast valid_until to bind it to a date input:

```php
protected $casts = [
        'valid_until' => 'date:Y-m-d',
    ];
```
All of these rules work, except the date rules.

Given valid_until is set to '2021-15-32' it passes date validation!
Given valid_until is set to '1970-01-01' it fails after_or_equal, but :input doesn't get replaced with the actual value.
![image](https://user-images.githubusercontent.com/12952240/115721407-4bed9400-a37e-11eb-9cfd-1cf47bfff101.png)

E-Mail for example works like expected and :input gets replaced with the actual value
![image](https://user-images.githubusercontent.com/12952240/115721843-b4d50c00-a37e-11eb-9900-a5dd97f11fc4.png)

**Context**
- Livewire version: 2.4.3
- Laravel version: 8.38.0

**Comment by ph7jack** [comment]
Can you check the return of the date input if it is in a valid format?
